### PR TITLE
Add integration tests for charindex, dateadd, to_timestamp (DNA-19468 / DNA-20817)

### PR DESCRIPTION
**Pull request by Dennis-UiPath**
## Description
- Adds integration tests for `charindex`, `dateadd` and `to_timestamp` macro's

## Release
- [X] Direct release (`main`)
- [ ] Merge to `dev` (or other) branch
  - Why:

### Did you consider?
- [X] Does it Work on Automation Suite / SQL Server
- [X] Does it Work on Automation Cloud / Snowflake
~~- [ ] What is the performance impact?~~
~~- [ ] The version number in `dbt_project.yml`~~

### CODE REVIEW

File: integration_tests/models/test_charindex.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+select
+    {# Find a substring (basic scenario) #}
+    {{ pm_utils.charindex('his', '`Text`') }} as `Find_basic_scenario`,
+    2 as `Find_basic_scenario_expected`,
+
+    {# Find a substring with multiple occurences #}
+    {{ pm_utils.charindex('is', '`Text`') }} as `Find_first_of_multiple_occurrences`,
+    3 as `Find_first_of_multiple_occurrences_expected`,
+
+    {# Find a space. SQL Server handles spaces differently #}
```
Comment (silviustanimir) [inline integration_tests/models/test_charindex.sql:15]: It's true that SQL Server handles spaces differently, but that's applicable for the `len` function, not the `charindex` function. Maybe rephrase this?

File: integration_tests/models/test_dateadd.sql [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    {{ pm_utils.to_varchar(pm_utils.dateadd('minute', 1, '`testdate`')) }} as `add_minutes`,
+    {{ pm_utils.to_varchar(pm_utils.dateadd('hour', 1, '`testdate`')) }} as `add_hours`,
+    {{ pm_utils.to_varchar(pm_utils.dateadd('day', 1, '`testdate`')) }} as `add_days`,
+    {{ pm_utils.to_varchar(pm_utils.dateadd('week', 1, '`testdate`')) }} as `add_weeks`,
+    {{ pm_utils.to_varchar(pm_utils.dateadd('month', 1, '`testdate`')) }} as `add_months`,
+    {{ pm_utils.to_varchar(pm_utils.dateadd('quarter', 1, '`testdate`')) }} as `add_quarters`,
+    {{ pm_utils.to_varchar(pm_utils.dateadd('year', 1, '`testdate`')) }} as `add_years`,
+
+    {# Use bigint as the value #}
+    {{ pm_utils.to_varchar(pm_utils.dateadd('year', '`bigint_value`', '`testdate`')) }} as `add_bigint`,
```
Comment (silviustanimir) [inline integration_tests/models/test_dateadd.sql:22]: No test for this use case?

File: integration_tests/models/test_dateadd.sql [hunk 3]
```diff
@@ -0,0 +1,51 @@
+with Input_data as (
+    select
+        '2023-11-12 13:14:15.678' as `testdate`,
+        null as `null_value`,
```
Comment (silviustanimir) [inline integration_tests/models/test_dateadd.sql:4]: This should be a datetime `NULL` value.

File: integration_tests/models/test_dateadd.sql [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        '2023-11-13 13:14:15.678' as `add_days_expected`,
+        '2023-11-19 13:14:15.678' as `add_weeks_expected`,
+        '2023-12-12 13:14:15.678' as `add_months_expected`,
+        '2024-02-12 13:14:15.678' as `add_quarters_expected`,
+        '2024-11-12 13:14:15.678' as `add_years_expected`,
+    {% endif %}
+    
+    {# Use null as the date #}
+    {{ pm_utils.to_varchar(pm_utils.dateadd('year', 1, '`null_value`')) }} as `add_to_null_value`,
+    '' as `add_to_null_value_expected`
```
Comment (silviustanimir) [inline integration_tests/models/test_dateadd.sql:49]: Then this should also be a `NULL` datetime/date? 🤔

File: integration_tests/models/test_to_timestamp.sql [hunk 5]
```diff
@@ -0,0 +1,44 @@
+with Input_data as (
+    select
+        '2023-11-12' as `timestamp_date`,
+        '2023-11-12 13:14:15' as `timestamp_datetime`,
+        '2023-11-12 13:14:15.678' as `timestamp_datetime_ms`,
+        '2023-11-12 13:14:15.6789' as `timestamp_datetime_ms4`,
+        '13:14:15' as `timestamp_time`,
+        null as `null_value`
```
Comment (silviustanimir) [inline integration_tests/models/test_to_timestamp.sql:8]: Same as before, shouldn't it be a datetime `NULL`?

**Review by silviustanimir — CHANGES_REQUESTED** [review]
Please address the comments, also check why `dbt deps` recursively creates folders indefinitely.

File: integration_tests/models/test_charindex.sql (hunk 1 above)
Comment (Dennis-UiPath) [reply integration_tests/models/test_charindex.sql]: Updated the comment to cover it better

File: integration_tests/models/test_dateadd.sql (hunk 2 above)
Comment (Dennis-UiPath) [reply integration_tests/models/test_dateadd.sql]: Added the testcase here

File: integration_tests/models/test_dateadd.sql (hunk 3 above)
Comment (Dennis-UiPath) [reply integration_tests/models/test_dateadd.sql:4]: Technically you are correct. We just removed the test for type tests from pm_utils. We can add it in the test and add Databricks support, but it is not compatible with Spark (no information_schema available). We discussed it and for now we will leave it this way, we will add a test if we run into an issue here.

**Review by silviustanimir — APPROVED** [review]
